blueprint/core/blueprint_ReactApplicationRoot.h

```diff
@@ -240,10 +250,23 @@ namespace blueprint
         {
             JUCE_ASSERT_MESSAGE_THREAD
 
+            // We early return here in the event that we're currently showing the red error
+            // screen. This prevents subsequent errors caused by dispatching events with an
+            // incorrect engine state from overwriting the first error message.
+            if (errorText)
```

Comment:
Cool so we're not going for a no-op here then? we just avoid the secondary error case. Meaning users need to be very careful about only dispatching events once AfterBundleEval is successful ? That's fine with me. In some ways it makes things a little more cumbersome from an API perspective, I could have various things listening to "engine" objects and dispatching events to JS in response. 

This is already sort of clear in the `ReactApplicationRoot::AfterBundleEval` docs

Reply:
Hm, I guess I thought this line implemented no-op behavior? That is, if you try to dispatch events while we know the engine to be in an error state (because we set the error text and it has not yet been cleared), we just early return (no-op).

If a user dispatches before the bundle eval, I suppose this will error... do you think we should avoid that case?

My thought was basically use the `afterBundleEval` to install initial state, then afterwards just dispatch events as usual.

Reply:
Yeah that's fine with me. It might just be one of those "be aware of X" things to note in the docs. It's reasonably clear from the `ReactApplicationRoot` header docs. I've no issues with it staying like this. I think as blueprint evolves and there are more community baked "binding" solutions (i.e. ValueTree, JSON patch etc.) this sort of concern might become less of an issue.

Reply:
Let's keep this one going in #92, because when that merges in it will break the current `subscribe` implementation. I'll mark this resolvd for now

Reply:
Whoops, meant for this comment to be on the above comment on the `subscribe` method!